test/evaluate/ISO-Fortran-binding.cc

```diff
@@ -210,14 +210,459 @@ int main() {
+        MATCH(true,
+            CFI_address(source, srcSubcripts) ==
+                CFI_address(result, resSubcripts));
+        MATCH(true,
+            CFI_address(source, srcSubcripts) ==
+                &array[k - lbs2][j - lbs1][i - lbs0]);
+        --resI;
+      }
+    }
+    ++resJ;
+  }
+}
+
+static void run_CFI_select_part_tests() {
+  constexpr std::size_t name_len = 5;
+  typedef struct {
+    double distance;
+    int stars;
+    char name[name_len];
+  } Galaxy;
+
+  const CFI_rank_t rank{2};
+  constexpr CFI_index_t universeSize[] = {2, 3};
+  Galaxy universe[universeSize[1]][universeSize[0]];
+
+  for (int j{0}; j < universeSize[1]; ++j) {
+    for (int i{0}; i < universeSize[0]; ++i) {
+      universe[j][i].distance = i + j * 32;
+      universe[j][i].stars = i * 2 + j * 64;
+      universe[j][i].name[2] = static_cast<char>(i);
+      universe[j][i].name[3] = static_cast<char>(j);
+    }
+  }
+
+  CFI_CDESC_T(rank) resStorage, srcStorage;
+  CFI_cdesc_t *result = reinterpret_cast<CFI_cdesc_t *>(&resStorage);
+  CFI_cdesc_t *source = reinterpret_cast<CFI_cdesc_t *>(&srcStorage);
+
+  bool testPreConditions{true};
+  int retCode = CFI_establish(result, nullptr, CFI_attribute_other,
+      CFI_type_int, sizeof(int), rank, nullptr);
+  testPreConditions &= (retCode == CFI_SUCCESS);
+  retCode = CFI_establish(source, &universe, CFI_attribute_other,
+      CFI_type_struct, sizeof(Galaxy), rank, universeSize);
+  testPreConditions &= (retCode == CFI_SUCCESS);
+  if (!testPreConditions) {
+    MATCH(true, testPreConditions);
+    return;
+  }
+
+  std::size_t displacement{offsetof(Galaxy, stars)};
+  std::size_t elem_len{0};  // ignored
+  retCode = CFI_select_part(result, source, displacement, elem_len);
+  MATCH(CFI_SUCCESS, retCode);
+
+  bool baseAddrShiftedOk =
+      (reinterpret_cast<char *>(source->base_addr) + displacement ==
+          result->base_addr);
+  MATCH(true, baseAddrShiftedOk);
+  if (!baseAddrShiftedOk) {
+    return;
+  }
+
+  MATCH(sizeof(int), result->elem_len);
+  for (CFI_index_t j{0}; j < universeSize[1]; ++j) {
+    for (CFI_index_t i{0}; i < universeSize[0]; ++i) {
+      CFI_index_t subscripts[]{
+          result->dim[0].lower_bound + i, result->dim[1].lower_bound + j};
+      MATCH(i * 2 + j * 64,
+          *reinterpret_cast<int *>(CFI_address(result, subscripts)));
+    }
+  }
+
+  // Test for Fortran character type
+  retCode = CFI_establish(
+      result, nullptr, CFI_attribute_other, CFI_type_char, 2, rank, nullptr);
+  testPreConditions &= (retCode == CFI_SUCCESS);
+  if (!testPreConditions) {
+    MATCH(true, testPreConditions);
+    return;
+  }
+
+  displacement = offsetof(Galaxy, name) + 2;
+  elem_len = 2;  // not ignored this time
+  retCode = CFI_select_part(result, source, displacement, elem_len);
+  MATCH(CFI_SUCCESS, retCode);
+
+  baseAddrShiftedOk =
+      (reinterpret_cast<char *>(source->base_addr) + displacement ==
+          result->base_addr);
```

Comment:
Does our f18 coding standard want bool typed expressions enclosed in outer parenthesis?

Reply:
I prefer to minimize unnecessary parentheses.

Reply:
Not that I know of.  Did `clang-format` add them?  Could it be configured to remove them?

Reply:
No it does not. [Google C++ style guide](https://google.github.io/styleguide/cppguide.html#Boolean_Expressions) says "_Feel free to insert extra parentheses judiciously because they can be very helpful in increasing readability when used appropriately_". I have to admit that this is probably not a judicious use. It is a personal habit, I will stop doing this.
Parenthesis removed.